src/fides/api/models/privacy_preference.py

```diff
@@ -72,296 +76,341 @@ class ServingComponent(Enum):
     tcf_banner = "tcf_banner"
 
 
-class LastSavedMixin:
-    """Stores common fields for the last saved preference or last served notice"""
+class ConsentIdentitiesMixin:
+    """Encrypted and hashed identities for consent reporting and last saved preference retrieval"""
+
+    email = Column(
+        StringEncryptedType(
+            type_in=String(),
+            key=CONFIG.security.app_encryption_key,
+            engine=AesGcmEngine,
+            padding="pkcs5",
+        ),
+    )  # Encrypted email
+
+    fides_user_device = Column(
+        StringEncryptedType(
+            type_in=String(),
+            key=CONFIG.security.app_encryption_key,
+            engine=AesGcmEngine,
+            padding="pkcs5",
+        ),
+    )  # Encrypted fides user device
+
+    phone_number = Column(
+        StringEncryptedType(
+            type_in=String(),
+            key=CONFIG.security.app_encryption_key,
+            engine=AesGcmEngine,
+            padding="pkcs5",
+        ),
+    )  # Encrypted phone number
+
+    hashed_email = Column(
+        String,
+        index=True,
+    )  # For exact match searches
+
+    hashed_fides_user_device = Column(String, index=True)  # For exact match searches
+
+    hashed_phone_number = Column(
+        String,
+        index=True,
+    )  # For exact match searches
+
+    @classmethod
+    def hash_value(
+        cls,
+        value: Optional[str],
+        encoding: str = "UTF-8",
+    ) -> Optional[str]:
+        """Utility function to hash the value with a generated salt
+        This returns None if there's no value, unlike ProvidedIdentity.hash_value
+        """
+        if not value:
+            return None
+
+        return ProvidedIdentity.hash_value(value, encoding)
+
+
+class CurrentPrivacyPreference(ConsentIdentitiesMixin, Base):
+    """Stores the latest saved privacy preferences for a given user
+
+    Email/phone/fides device must be unique.  If we later tie identities together, these records
+    are consolidated.
+    """
+
+    @declared_attr
+    def __tablename__(self) -> str:
+        """Table name is currentprivacypreferencev2 - currentprivacypreference was deprecated and removed"""
+        return "currentprivacypreferencev2"
```

Comment:
Renaming CurrentPrivacyPreferenceV2 and LastServedNoticeV2 models to CurrentPrivacyPreference and LastServedNotice now that we don't need to maintain the v2 distinction.  The tablename is still v2 though -